src/components/tests/SourcesTree.spec.js

```diff
@@ -6,32 +6,617 @@ import React from "react";
+        sourceId: "server1.conn13.child1/39"
+      });
+    });
+
+    it("should not select item with children", async () => {
+      defaultComponent.instance().selectItem(mockDirectory);
+      expect(defaultProps.selectLocation).not.toHaveBeenCalled();
+    });
+
+    it("should select item on enter", async () => {
+      await defaultComponent.instance().focusItem(singleMockItem);
+      await defaultComponent.update();
+      await defaultComponent
+        .find(".sources-list")
+        .simulate("keydown", { keyCode: 13 });
+      expect(defaultProps.selectLocation).toHaveBeenCalledWith({
+        sourceId: "server1.conn13.child1/39"
+      });
+    });
+
+    it("does not select if no item is focused on", async () => {
+      await defaultComponent
+        .find(".sources-list")
+        .simulate("keydown", { keyCode: 13 });
+      expect(defaultProps.selectLocation).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("handles items", () => {
+    it("getChildren from directory", async () => {
+      const item = createMockItem("http://mdn.com/views", "views", ["a", "b"]);
+      const children = defaultComponent
+        .find("ManagedTree")
+        .props()
+        .getChildren(item);
+      expect(children).toEqual(["a", "b"]);
+    });
+
+    it("getChildren from non directory", async () => {
+      const children = defaultComponent
+        .find("ManagedTree")
+        .props()
+        .getChildren(singleMockItem);
+      expect(children).toEqual([]);
+    });
+
+    it("onExpand", async () => {
+      const expandedState = ["x", "y"];
+      await defaultComponent
+        .find("ManagedTree")
+        .props()
+        .onExpand({}, expandedState);
+      expect(defaultProps.setExpandedState).toHaveBeenCalledWith(expandedState);
+    });
+
+    it("onCollapse", async () => {
+      const expandedState = ["y", "z"];
+      await defaultComponent
+        .find("ManagedTree")
+        .props()
+        .onCollapse({}, expandedState);
+      expect(defaultProps.setExpandedState).toHaveBeenCalledWith(expandedState);
+    });
+
+    it("getParent", async () => {
+      const item = defaultComponent.state("sourceTree").contents[0].contents[0];
+      const parent = defaultComponent
+        .find("ManagedTree")
+        .props()
+        .getParent(item);
+
+      expect(parent.path).toEqual("mdn.com");
+      expect(parent.contents).toHaveLength(3);
+    });
+  });
+
+  describe("getPath", () => {
+    it("should return path for item", async () => {
+      const path = defaultComponent.instance().getPath(singleMockItem);
+      expect(path).toEqual("http://mdn.com/one.js/one.js/");
+    });
+
+    it("should return path for blackboxedboxed item", async () => {
+      const path = blackboxedComponent.instance().getPath(blackboxedMockItem);
+      expect(path).toEqual("http://mdn.com/blackboxed.js/blackboxed.js/update");
+    });
+  });
+
+  describe("test redux connect", () => {
```

Comment:
Yeah, we don't need to get the coverage here, it is okay not to have 100% coverage :)